app/channel.go

```diff
@@ -2622,10 +2704,32 @@ func (a *App) MarkChannelsAsViewed(channelIDs []string, userID string, currentSe
 	for _, channelID := range channelsToClearPushNotifications {
 		a.clearPushNotification(currentSessionId, userID, channelID)
 	}
+
+	if !collapsedThreadsSupported {
+		// for compatibility with old clients, when channel is viewed - mark all threads in that channel as read
```

Comment:
I see this "checking if threads are enabled" happen a few time, every time spelled out. Worth extracting into a function? Also seems to happen at line 2456 and perhaps more places>

Reply:
Do you mean move all the code for `!collapsedThreadsSupported` into a separate functions?

Reply:
I mean the code to check "is collapsed threads switched on for this user?"